### ISSUE TITLE
Using dokkatoo in precompiled script plugins causes "Internal Dokkatoo API" warnings

### ISSUE THREAD

**Issue by translatenix** [adamko-dev/dokkatoo#176]
Gradle 8.6, Dokkatoo 2.1.0
```
> Task :buildSrc:compileKotlin
w: file:///home/me/prj/buildSrc/build/generated-sources/kotlin-dsl-accessors/kotlin/gradle/kotlin/dsl/accessors/_449187fc874922a2ad1c7ba7612ed738/Accessorsd6bujp36mck8x4gylrvrl8096.kt:65:103 Internal Dokkatoo API - may change at any time without notice
w: file:///home/me/prj/buildSrc/build/generated-sources/kotlin-dsl-accessors/kotlin/gradle/kotlin/dsl/accessors/_449187fc874922a2ad1c7ba7612ed738/Accessorsd6bujp36mck8x4gylrvrl8096.kt:66:152 Internal Dokkatoo API - may change at any time without notice
w: file:///home/me/prj/buildSrc/build/generated-sources/kotlin-dsl-accessors/kotlin/gradle/kotlin/dsl/accessors/_449187fc874922a2ad1c7ba7612ed738/Accessorsd6bujp36mck8x4gylrvrl8096.kt:72:120 Internal Dokkatoo API - may change at any time without notice
w: file:///home/me/prj/buildSrc/build/generated-sources/kotlin-dsl-accessors/kotlin/gradle/kotlin/dsl/accessors/_449187fc874922a2ad1c7ba7612ed738/Accessorsd6bujp36mck8x4gylrvrl8096.kt:73:64 Internal Dokkatoo API - may change at any time without notice
w: file:///home/me/prj/buildSrc/build/generated-sources/kotlin-dsl-accessors/kotlin/gradle/kotlin/dsl/accessors/_449187fc874922a2ad1c7ba7612ed738/Accessorsd6bujp36mck8x4gylrvrl8096.kt:73:116 Internal Dokkatoo API - may change at any time without notice
```

**Comment by aSemy** [comment]
Hi, thanks for the report.

The log messages by themselves don't provide enough info as I can't see what code is triggering the message. Can you share the convention plugin? 

Alternatively, you can opt-in to the API, but that of course risks changes.

**Comment by translatenix** [comment]
Here is a minimal reproducer: https://github.com/translatenix/dokkatoo-warnings

How do I opt in? The warnings aren't useful.

**Comment by aSemy** [comment]
Huh, weird! Thanks for the report, that's really unexpected. Dokkatoo itself has a similar convention plugin and I've not seen those same warnings. I'll have a closer look.

There are instructions for opting-in on the [Kotlin docs page](https://kotlinlang.org/docs/opt-in-requirements.html), but I'll try and fix it so they don't appear. Aside from opting-in, you can just ignore the warnings for now.

**Comment by aSemy** [comment]
This will be fixed in v2.2.0, thanks again for the report.
